questions.md

```diff
@@ -0,0 +1 @@
+Can you explain the -set-upsteam option for git push?
```

Comment:
:wave: Hi @TeacherTalley ! Great question! This is essentially a shortcut that allows you to establish the relationship between your local and remote branch. For example:

If I have a branch on my computer called `hello-world` and I have a remote branch on GitHub that is also called `hello-world` I can set that these two branches are related by typing the following while in my local hello-world branch:

`git branch --set-upstream hello-world` "hello-world" in this case is the name of my remote branch

So, in the future you no longer have to type:

`git push origin hello-world`

because you have established that all future `push`es are going to the remote branch `hello-world` and you can just type `git push` instead. 

So, `--set-upstream` is essentially telling Git that if I use any git command in the future on this branch (`git pull`, `git push`, `git fetch`, etc.) Git will know that you are asking it to do this in relation to the remote branch. It is a little bit of a shortcut that helps you to type less.